File: ios/ui-framework-sample/UIFramework/Sources/UIFramework/BizCalendarSheet.swift
import SwiftUI

public struct BizCalendarSheetModifier: ViewModifier {
    @Binding var isPresented: Bool
    var title: String?
    var config: CalendarConfig
    var initialDate: Date?
    var showActionButtons: Bool
    var onDateSelected: ((CalendarSelection) -> Void)?
    var onRangeSelected: ((CalendarSelection?, CalendarSelection?) -> Void)?
    var onMonthChanged: ((Int, Int) -> Void)?
    var onConfirm: ((CalendarSelection?, CalendarSelection?) -> Void)?
    var onCancel: (() -> Void)?
    
    public func body(content: Content) -> some View {
        content.bizBottomSheet(
            isPresented: $isPresented,
            header: {
                if let title = title {
                    Text(title)
                        .font(.system(size: 16, weight: .bold))
                        .foregroundColor(Color(hex: "#1A2A5E"))
                        .padding(.top, 8)
                } else {
                    EmptyView()
                }
            },
            footer: { EmptyView() },
            content: {
                BizCalendar(
                    config: config,
                    initialDate: initialDate ?? Date(),
                    showActionButtons: showActionButtons,
                    onDateSelected: onDateSelected,
                    onRangeSelected: onRangeSelected,
                    onMonthChanged: onMonthChanged,
                    onConfirm: { start, end in
                        onConfirm?(start, end)
                        isPresented = false
                    },
                    onCancel: {
                        onCancel?()
                        isPresented = false
                    }
                )
                .padding(.bottom, 24)
            }
        )
    }
}

public extension View {
    func bizCalendarSheet(
        isPresented: Binding<Bool>,
        title: String? = nil,
        config: CalendarConfig = CalendarConfig(),
        initialDate: Date? = nil,
        showActionButtons: Bool = true,
        onDateSelected: ((CalendarSelection) -> Void)? = nil,
        onRangeSelected: ((CalendarSelection?, CalendarSelection?) -> Void)? = nil,
        onMonthChanged: ((Int, Int) -> Void)? = nil,
        onConfirm: ((CalendarSelection?, CalendarSelection?) -> Void)? = nil,
        onCancel: (() -> Void)? = nil
    ) -> some View {
        self.modifier(BizCalendarSheetModifier(
            isPresented: isPresented,
            title: title,
            config: config,
            initialDate: initialDate,
            showActionButtons: showActionButtons,
            onDateSelected: onDateSelected,
            onRangeSelected: onRangeSelected,
            onMonthChanged: onMonthChanged,
            onConfirm: onConfirm,
            onCancel: onCancel
        ))
    }
}

File: android/ui-framework-sample/ui-framework/src/main/java/kh/mcnc/lib/components/BizCalendarSheet.kt
package kh.mcnc.lib.components

import androidx.compose.runtime.Composable
import kh.mcnc.lib.utils.CalendarConfig
import java.util.Date

@Composable
fun BizCalendarSheet(
    visible: Boolean,
    onDismissRequest: () -> Unit,
    title: String? = null,
    config: CalendarConfig? = null,
    initialDate: Date? = null,
    showActionButtons: Boolean = true,
    onDateSelected: ((Any) -> Unit)? = null,
    onRangeSelected: ((Any?, Any?) -> Unit)? = null,
    onMonthChanged: ((Int, Int) -> Unit)? = null,
    onConfirm: ((Any?, Any?) -> Unit)? = null,
    onCancel: (() -> Unit)? = null
) {
    BizBottomSheet(
        isVisible = visible,
        onDismissRequest = onDismissRequest,
        
    ) {
        BizCalendar(
            config = config ?: CalendarConfig(),
            initialDate = initialDate ?: Date(),
            showActionButtons = showActionButtons,
            onDateSelected = onDateSelected,
            onRangeSelected = onRangeSelected,
            onMonthChanged = onMonthChanged,
            onConfirm = onConfirm,
            onCancel = onCancel
        )
    }
}
